src/Features/CSharp/Portable/ChangeSignature/CSharpChangeSignatureService.cs

```diff
@@ -411,29 +460,100 @@ private SyntaxNode GetNodeContainingTargetNode(SyntaxNode matchingNode)
+                    {
+                        if (addedParameter.CallsiteValue != null)
+                        {
+                            fullList.Add(SyntaxFactory.Argument(
+                                seenNameEquals ? SyntaxFactory.NameColon(addedParameter.Name) : default,
+                                refKindKeyword: default,
+                                expression: SyntaxFactory.ParseExpression(addedParameter.CallsiteValue)));
+                            separators.Add(SyntaxFactory.Token(SyntaxKind.CommaToken).WithTrailingTrivia(SyntaxFactory.ElasticSpace));
+                        }
+                    }
+                    else
+                    {
+                        if (indexInExistingList < newArguments.Count)
+                        {
+                            if (newArguments[indexInExistingList].NameColon != default)
+                            {
+                                seenNameEquals = true;
+                            }
 
-                // copy whitespace trivia from original position
-                param = TransferLeadingWhitespaceTrivia(param, list[index]);
+                            if (indexInExistingList < newArguments.SeparatorCount)
+                            {
+                                separators.Add(newArguments.GetSeparator(indexInExistingList));
+                            }
 
-                newParameters.Add(param);
+                            fullList.Add(newArguments[indexInExistingList++]);
+                        }
+                    }
+                }
             }
 
-            var numSeparatorsToSkip = originalParameters.Count - reorderedParameters.Count;
-            return SyntaxFactory.SeparatedList(newParameters, GetSeparators(list, numSeparatorsToSkip));
+            // Add the rest of existing parameters, e.g. from the params argument.
+            while (indexInExistingList < newArguments.Count)
+            {
+                if (indexInExistingList < newArguments.SeparatorCount)
+                {
+                    separators.Add(newArguments.GetSeparator(indexInExistingList));
+                }
+
+                fullList.Add(newArguments[indexInExistingList++]);
+            }
+
+            if (fullList.Count == separators.Count && separators.Count != 0)
+            {
+                separators.Remove(separators.Last());
+            }
+
+            return SyntaxFactory.SeparatedList(fullList, separators);
+        }
+
+        private static ParameterSyntax CreateNewParameterSyntax(AddedParameter addedParameter)
+            => CreateNewParameterSyntax(addedParameter, skipType: false);
+
+        private static ParameterSyntax CreateNewParameterSyntax(AddedParameter addedParameter, bool skipType)
+            => SyntaxFactory.Parameter(
+                attributeLists: SyntaxFactory.List<AttributeListSyntax>(),
+                modifiers: SyntaxFactory.TokenList(),
```

Comment:
can we not use existing code-gen features to do this?

Reply:
https://github.com/dotnet/roslyn/issues/41130